[webpack] Reports and report-dependent code

orangejenny opened the pull request:
## Technical Summary
This moves standard reports, UCR, admin reports, microplanning reports, other one-off reports (e.g., linked project space history and repeat records), custom reports, data interfaces, the export download page, and the view lookup table page to webpack. It's a mixture of areas that were already on requirejs and areas that were still using script tags.

I've been holding off on PRing this until some work extracting the inddex reports' inline js into a separate file was complete (https://github.com/dimagi/commcare-hq/pull/35586). I'm still working through some bugs in that work, which is why this PR is marked don't merge, but decided it's time to open this PR for review. QA is already complete.

## Safety Assurance

### Safety story
This is fairly risky from a UI perspective, with the risk being disabling one or more of these UIs. There's no deeper risk to data, etc. QA has done extensive testing.

### Automated test coverage

Not much.

### QA Plan

https://dimagi.atlassian.net/browse/QA-7353

### Rollback instructions

This is a big PR that probably won't roll back cleanly for very long, but there's no technical reason not to roll it back.

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

orangejenny committed:
Added missing dependency to reports/js/filters/schedule_instance (cherry pick)

orangejenny committed:
Migrated submission_error_report.html

orangejenny committed:
Added explicit dependencies to reports/js/charts/*

orangejenny committed:
Added explicit dependencies to reports_core/*

orangejenny committed:
Updated base.js

orangejenny committed:
Updated maps_utils.js

orangejenny committed:
Updated standard_hq_report to support webpack

orangejenny committed:
Added dependency for fadeVisible used in drilldown (cherry pick)

orangejenny committed:
Migrated download export page

orangejenny committed:
Migrated view lookup tables page to webpack

The non-minified version of datatables errors out.
This is the B3 library so I'm not going to worry about it.
Webpack will minify it.

orangejenny committed:
Added daterangepicker dependency to filters/main.js (cherry pick)

orangejenny committed:
Added entry point to base_template.html, which is the template for some reports

orangejenny committed:
Updated geospatial reports

orangejenny committed:
Migrated repeat record report

orangejenny committed:
Removed NVD3 and D3 from base template

These should now be included directly in js files where they're needed.

orangejenny committed:
Migrated tableau

orangejenny committed:
Exclude node_modules from AMD-related webpack loaders (cherry pick)

orangejenny committed:
Removed unused code from standard_hq_report.js

orangejenny committed:
Migrated standard reports

orangejenny committed:
Added explicit dependencies in case list explorer

orangejenny committed:
Migrated custom inddex reports

orangejenny committed:
Migrated custom mc report

orangejenny committed:
Migrated userreports/reportbuilder/data_source_select.html

orangejenny committed:
Migrated userreports/reportbuilder/configure_report.html

orangejenny committed:
Migrated userreports/configurable_report.html

orangejenny committed:
Removed script tags from userreports/base.html now that all descendants are migrated

orangejenny committed:
Migrated reports_home.html

millerdev reviewed (approved):
It's wonderful to have gotten this far!

orangejenny commented:
@biyeun Totally forgot about diffs, thanks, I'll add those.

orangejenny committed:
Bootstrap 5 Migration - Rebuilt diffs

orangejenny commented:
There's also a real test failure, which is intriguing, I'll check that out.

orangejenny committed:
Fixed userreports js tests

These tests aren't using bundling, which means they're dependent on the ordering of script tags.

orangejenny committed:
Merge branch 'master' into jls/webpack-reports-dependencies

biyeun commented:
@orangejenny had an issue loading submissions by form report. Looks like fixed columns is missing

orangejenny committed:
Added datatables.fixedColumns to datatables_config.js

orangejenny committed:
Updated datatables.fixedColumns alias

orangejenny commented:
@biyeun Thank you, I'm glad you found that before this went live. Fixed in https://github.com/dimagi/commcare-hq/pull/35599/commits/89626816624bbc8546ddd868cf26c66ac09ebf50 + https://github.com/dimagi/commcare-hq/pull/35599/commits/f4fb700e4304e90dbc1513dbb087429a56cf9539
